zephyr/samples/profiles/b-rtr/src/main.c

```diff
@@ -0,0 +1,185 @@
+#include "bacnet/basic/object/trendlog.h"
+#if defined(INTRINSIC_REPORTING)
+#include "bacnet/basic/object/nc.h"
+#endif /* defined(INTRINSIC_REPORTING) */
+
+/* Logging module registration is already done in ports/zephyr/main.c */
+#include <logging/log.h>
+LOG_MODULE_DECLARE(bacnet, CONFIG_BACNETSTACK_LOG_LEVEL);
+
+/** Buffer used for receiving */
+static uint8_t Rx_Buf[MAX_MPDU] = { 0 };
+
+#define VIRTUAL_DNET  2709      /* your choice of number here */
+
+int DNET_list[2] = {
+    VIRTUAL_DNET, -1 /* Need -1 terminator */
+};
+
+/* routed devices - I-Am on startup */
+//static unsigned Routed_Device_Index;
+
+/** Initialize the handlers we will utilize.
+ * @see Device_Init, apdu_set_unconfirmed_handler, apdu_set_confirmed_handler
+ */
+static void service_handlers_init(void)
+{
+	Device_Init(NULL);
+	/* we need to handle who-is to support dynamic device binding */
+	/* spec. K.5.2 DM-DDB-B  */
+	apdu_set_unconfirmed_handler(SERVICE_UNCONFIRMED_WHO_IS,
+				     handler_who_is);
+	/* spec. K.5.4 DM-DOB-B  */
+	apdu_set_unconfirmed_handler(SERVICE_UNCONFIRMED_WHO_HAS,
+				     handler_who_has);
+	/* set the handler for all the services we don't implement */
+	/* It is required to send the proper reject message... */
+	apdu_set_unrecognized_service_handler_handler(
+		handler_unrecognized_service);
+	/* Set the handlers for any confirmed services that we support. */
+	/* We must implement read property - it's required! */
+	/* spec. K.1.2 DM-RP-B  */
+	apdu_set_confirmed_handler(SERVICE_CONFIRMED_READ_PROPERTY,
+				   handler_read_property);
+//	apdu_set_confirmed_handler(SERVICE_CONFIRMED_READ_PROP_MULTIPLE,
+//				   handler_read_property_multiple);
+	/* spec. K.1.6 DM-WP-B  */
+	apdu_set_confirmed_handler(SERVICE_CONFIRMED_WRITE_PROPERTY,
+				   handler_write_property);
+//	apdu_set_confirmed_handler(SERVICE_CONFIRMED_WRITE_PROP_MULTIPLE,
+//				   handler_write_property_multiple);
+//	apdu_set_confirmed_handler(SERVICE_CONFIRMED_REINITIALIZE_DEVICE,
+//				   handler_reinitialize_device);
+	/* handle communication so we can shutup when asked */
+//	apdu_set_confirmed_handler(
+//		SERVICE_CONFIRMED_DEVICE_COMMUNICATION_CONTROL,
+//		handler_device_communication_control);
+
+	/* spec. K.6.4 NM-RC-B  */
+/*
+Who-Is-Router-To-Network 	i e
+I-Am-Router-To-Network 		i e
+Reject-Message-To-Network  	i e
+Router-Busy-To-Network 		i e
+Router-Available-To-Network i e
+Network-Number-Is 			i e
+What-Is-Network-Number 		  e
+*/
+}
+
+void main(void)
+{
+	LOG_INF("\n*** BACnet Profile B-RTR Sample ***\n");
+	LOG_INF("BACnet Stack Version " BACNET_VERSION_TEXT);
+	LOG_INF("BACnet Device ID: %u", Device_Object_Instance_Number());
+	LOG_INF("BACnet Device Max APDU: %d", MAX_APDU);
+
+	service_handlers_init();
+	datalink_init(NULL);
+
+	/* configure the timeout values */
+	int64_t last_ms = k_uptime_get();
+
+	/* broadcast an I-Am on startup */
+	Send_I_Am(&Handler_Transmit_Buffer[0]);
+
+    /* broadcast an I-am-router-to-network on startup */
```

Comment:
It looks like leading whitespace is inconsistent (tabs/spaces) with other lines in the function (file?).  Please fix.